core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -196,28 +196,44 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       case NonFatal(e) => log.warn(e.getMessage, e)
     }
 
+    val hiveTables = Misc.getMemStore.getExternalCatalog.getHiveTables(true).asScala
+
     try {
       // External Tables
-      val hiveTables: java.util.List[ExternalTableMetaData] =
-        Misc.getMemStore.getExternalCatalog.getHiveTables(true)
-      externalTables = hiveTables.asScala.collect {
-        case table if table.tableType.equalsIgnoreCase("EXTERNAL") =>
+      externalTables = hiveTables.collect {
```

Comment:
externalTables need not be a var. Can be a val if we are not changing it

Reply:
Done. Also externalTables initialization moved inside block.